scripts/release.sh

```diff
@@ -162,14 +163,16 @@ main() {
     fi
 
     # Verify the latest commit has the version tag
-    local tag="$(git describe --exact-match HEAD)"
+    tag="$(git describe --exact-match HEAD)"
+    local tag
```

Comment:
any reason why you break the one line `local tag="$(git describe --exact-match HEAD)"` into two lines? 

After breaking into two lines, then it has two issues:
1. The local variable's value is empty;
2. It will have impact on the same variable name outside the scope. 

See example below,

```
#!/usr/bin/env bash

myid="id_outside"

demo() {
  myid="id_inside"
  local myid
  echo "inside, ${myid}"
}

demo
echo "outside2, $myid"
```

The output is,
```
inside, 
outside2, id_inside
```

Reply:
This is shellcheck warning that mentions that `local tag="$(git describe --exact-match HEAD)"` hides status from `git describe --exact-match HEAD`.

Reply:
Decided to silence all shellcheck warnings. During v3.5.2 I didn't encounter any problems so I don't think fixing those warnings will bring any value. On the other hand making any changes without testing it properly could introduce errors.

Removal of those silences should be done in a separate PR, by someone that is more knowledgeable about bash.